tests/ui/macros/vec-macro-in-pattern.stderr

```diff
@@ -1,10 +1,13 @@
-error: arbitrary expressions aren't allowed in patterns
+error: expected pattern, found `#`
   --> $DIR/vec-macro-in-pattern.rs:7:14
    |
 LL |         Some(vec![43]) => {}
    |              ^^^^^^^^
+   |              |
+   |              expected pattern
+   |              in this macro invocation
+   |              this macro call doesn't expand to a pattern
```

Comment:
hmm... does the compiler know what it expands to?

Reply:
I don't understand the question. Are you suggesting changing the error message?

Reply:
well, I was wondering if we even have the information reliably to change the error message to "this macro call expands to an expression, not a pattern"

Reply:
Let's see... the error is emitted by `annotate_err_with_kind`, which is called in this case from `emit_frag_parse_err`, which is called from `ParserAnyMacro::make` after the call to `parse_ast_fragment` fails.

I think it's hard to do what you suggest, we try to parse as a pattern and when that fails there's no easy way to determine what we actually had instead of a pattern, short of reparsing it in a bunch of different ways to see if anything succeeds, which feels clumsy.

Reply:
yea no, that's wrong. Thanks for checking